sdk/src/store.rs

```diff
@@ -1468,39 +1575,52 @@ impl Store {
     fn start_save(
         &mut self,
         asset_path: &Path,
-        output_path: &Path,
+        dest_path: &Path,
         reserve_size: usize,
     ) -> Result<Vec<u8>> {
         // clone the source to working copy if requested
+
         get_supported_file_extension(asset_path).ok_or(Error::UnsupportedType)?; // verify extensions
-        let ext = get_supported_file_extension(output_path).ok_or(Error::UnsupportedType)?;
-        if asset_path != output_path {
-            fs::copy(&asset_path, &output_path).map_err(Error::IoError)?;
+        let ext = get_supported_file_extension(dest_path).ok_or(Error::UnsupportedType)?;
+        if asset_path != dest_path {
+            fs::copy(&asset_path, &dest_path).map_err(Error::IoError)?;
         }
 
         //  update file following the steps outlined in CAI spec
 
         // 1) Add DC provenance XMP
-        // update XMP info & add xmp hash to provenance claim
-        #[cfg(feature = "xmp_write")]
-        if let Some(provenance) = self.provenance_path() {
-            embedded_xmp::add_manifest_uri_to_file(output_path, &provenance)?;
-        } else {
-            return Err(Error::XmpWriteError);
-        }
+        let pc = self.provenance_claim().ok_or(Error::ClaimEncoding)?;
+        let output_path = match pc.remote_manifest() {
+            crate::claim::RemoteManifest::NoRemote => {
+                // update XMP info & add xmp hash to provenance claim
+                #[cfg(feature = "xmp_write")]
+                if let Some(provenance) = self.provenance_path() {
+                    embedded_xmp::add_manifest_uri_to_file(dest_path, &provenance)?;
+                } else {
+                    return Err(Error::XmpWriteError);
+                }
+                dest_path.to_path_buf()
+            }
+            crate::claim::RemoteManifest::SideCar => dest_path.with_extension(MANIFEST_STORE),
+            crate::claim::RemoteManifest::Remote(url) => {
+                let d = dest_path.with_extension(MANIFEST_STORE);
+                embedded_xmp::add_manifest_uri_to_file(dest_path, &url)?;
```

Comment:
I think you need the feature= "xmp_write" here with a failure if it is not enabled.

Reply:
Interesting add_manifest_uri_to_file is not behind that feature flag

Reply:
The entire embeded_xmp module is behind the "xmp-write" flag. I've verified that this won't build with if that flag is disabled.
```
cd sdk && cargo test --features="file_io" --no-default-features 
   Compiling c2pa v0.11.3 (/Users/gpeacock/dev/c2pa-rs/sdk)
error[E0433]: failed to resolve: use of undeclared crate or module `embedded_xmp`
    --> sdk/src/store.rs:1614:17
     |
1614 |                 embedded_xmp::add_manifest_uri_to_file(dest_path, &url)?;
     |                 ^^^^^^^^^^^^ use of undeclared crate or module `embedded_xmp`

```

Reply:
Done